include/mqt-core/QuantumComputation.hpp

```diff
@@ -37,6 +37,7 @@ class QuantumComputation {
       typename std::vector<std::unique_ptr<Operation>>::const_iterator;
 
   friend class CircuitOptimizer;
+  friend class DDMinimizer;
```

Comment:
Hey there 👋🏼 

Many thanks for opening this draft PR.
I just wanted to add a quick point here: I'd be rather hesitant to add a `friend class` declaration here to a class that is not even defined within this project.

Is the public API of the `QuantumComputation` class not enough to accomplish what you want?
If it isn't, then I'd rather rethink the public API instead of adding friend classes with "unlimited access".